src/components/App/App.component.jsx

```diff
@@ -1,17 +1,31 @@
-import React, { useLayoutEffect } from 'react';
+import React, { useLayoutEffect, useReducer, useState } from 'react';
 import { BrowserRouter, Switch, Route } from 'react-router-dom';
 
 import AuthProvider from '../../providers/Auth';
 import HomePage from '../../pages/Home';
 import LoginPage from '../../pages/Login';
 import NotFound from '../../pages/NotFound';
-import SecretPage from '../../pages/Secret';
-import Private from '../Private';
-import Fortune from '../Fortune';
-import Layout from '../Layout';
+import VideoDetail from '../../pages/VideoDetail';
+import FavoriteVideos from '../../pages/FavoriteVideos';
+// import Private from '../Private';
+// import Layout from '../Layout';
+import VideoReducer from '../../store/VideoReducer';
+import VideoContext from '../../store/VideoContext';
+import Navbar from '../Navbar';
+
 import { random } from '../../utils/fns';
 
 function App() {
+  const [inputState, setInputState] = useState();
+
+  const [state, dispatch] = useReducer(VideoReducer, {
+    currentVideo: {},
+    favoriteVideos: [],
+    videos: [],
+    setCurrentVideo: () => {},
```

Comment:
Reducers shouldn't return functions. If you want to include those functions to the state, please concat them with the state that ```useReducer``` returns. Also I think that it'd be better if you create the Provider as a new component something like this: 
```javascript
const initialState = {  
  currentVideo: {},
  favoriteVideos: [],
  videos: [],
};

function VideosProvider({ children }) {
  const [{ currentVideo, ... }, dispatch] = useReducer(VideoReducer, initialState);

  const state = {
    currentVideo,
    ...
    setCurrentVideo: () => {},
    ...
  };

  return (
     <VideoContext.Provider value={{ state={...state, 
  )

}
```